subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1084,9 +1090,48 @@ public void preventFromFurtherMutation() {
             configurationAttributes = new ImmutableAttributeContainerWithErrorMessage(delegatee, this.displayName);
             outgoing.preventFromFurtherMutation();
             canBeMutated = false;
+
+            // If capabilities or attributes are empty, duplicate attribute sets are allowed, else check
+            if (isCanBeConsumed() && !getOutgoing().getCapabilities().isEmpty() && !getAttributes().isEmpty()) {
```

Comment:
Your code uses `&&` but the comment says "or" ... and I agree this should be _and_

Reply:
I'll rephrase the comment to be more clear.